addons/mail/static/src/messaging/component/activity_mark_done_popover/activity_mark_done_popover_tests.js

```diff
@@ -0,0 +1,278 @@
+        id: 12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        "Popover component should be present"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_feedback',
+        "Popover component should contain the feedback textarea"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_buttons',
+        "Popover component should contain the action buttons"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneScheduleNextButton',
+        "Popover component should contain the done & schedule next button"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ActivityMarkDonePopover_doneButton',
+        "Popover component should NOT contain the done button"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ActivityMarkDonePopover_discardButton',
+        "Popover component should NOT contain the discard button"
+    );
+});
+
+QUnit.test('activity mark done popover click on discard', async function (assert) {
+    assert.expect(4);
+
+    await this.start();
+    const activity = this.env.entities.Activity.create({
+        activity_category: 'not_upload_file',
+        can_write: true,
+        id: 12,
+    });
+    await this.createActivityMarkDonePopoverComponent(activity);
+    document.addEventListener('o-discard-clicked', () => {
+        assert.step('event_triggered');
+    }, true);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover',
+        "Popover component should be present"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_ActivityMarkDonePopover_discardButton',
+        "Popover component should contain the discard button"
+    );
+    document.querySelector('.o_ActivityMarkDonePopover_discardButton').click();
+    assert.verifySteps(['event_triggered'], 'Discard clicked should trigger the right event');
+
+});
+
+QUnit.test('activity mark done popover mark done without feedback', async function (assert) {
+    assert.expect(7);
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (route === '/web/dataset/call_kw/mail.activity/action_feedback') {
+                assert.step('action_feedback');
+                assert.strictEqual(args.args.length, 1);
+                assert.strictEqual(args.args[0].length, 1);
+                assert.strictEqual(args.args[0][0], 12);
+                assert.strictEqual(args.kwargs.attachment_ids.length, 0);
+                assert.notOk(args.kwargs.feedback);
+                return;
+            }
+            if (route === '/web/dataset/call_kw/mail.activity/unlink') {
+                // it was called before producing a "cannot find record" error
+                throw new Error("unlink rpc method should not be called");
```

Comment:
I would further explain error message, like _"'unlink' RPC on activity must not be called (already unlinked from mark as done)"_, so that it's easier to tell what's wrong when we have this error in the QUnit interface.

It's good to have a comment that further explain why this must be treated as an error, but the comment is not well formulated IMO. Something like _"'unlink' on non-existing record raises a server crash"_ is much better.